### DISCUSSION TITLE
Filtered view of running netlab nodes (clab) based on working directory

### DISCUSSION THREAD

**Discussion by sdktr** [ipspace/netlab#1008] [Q&A]
Using containerlab as provider:
When running in multilab mode (but also if other docker images are running), the output of ```netlab status``` is cluttered with images not relevant to the lab we're interested in.

Since there already is an assumption of having the netlab state info available in the working directory, I propose we use that to filter the output of netlab status by default to only include the current lab nodes.

What I'm not aware of, is whether this can also be reused for non-clab backends. I'm just now noticing the ```instance``` parameter to the netlab CLI, will look into that.

**Comment by sdktr** [comment]
> Using containerlab as provider: When running in multilab mode (but also if other docker images are running), the output of `netlab status` is cluttered with images not relevant to the lab we're interested in.
> 
> Since there already is an assumption of having the netlab state info available in the working directory, I propose we use that to filter the output of netlab status by default to only include the current lab nodes.
> 
> What I'm not aware of, is whether this can also be reused for non-clab backends. I'm just now noticing the `instance` parameter to the netlab CLI, will look into that.

I guess this is already implemented but I'm not using it correctly..
https://github.com/ipspace/netlab/blob/ab2440b10252fe5a2125c530216aaa84f927888e/netsim/cli/status.py#L76

I'm unable to get a filtered view when using clab as the provider and multilab plugin.

**Comment by ipspace** [comment]
There are two aspects to this:

* What internal information **netlab status** displays. That's controlled with the **instance** parameter in the **netlab status show** command
* What workload information is displayed in **netlab status [ list ].** That's fixed and is (IIRC) either **virsh list** or **docker ps**

My primary focus when writing **netlab status** was to identify stale/forgotten lab instances. I have no problem changing the default to display information about current lab, maybe using **containerlab status** and **vagrant status** to limit the display to the current instance, and having '--all' flag to display all labs/workloads, or adding workload display to **netlab status show instance**, potentially adding a special **current** instance to refer to the current directory.

What would you find least confusing? Which way would you like to go? /cc @petercrocker @ssasso @jbemmel

**Comment by ipspace** [comment]
Any preferences @sdktr @petercrocker @ssasso @jbemmel @steffann? If not, I'll just go ahead and implement what makes sense to me (but don't complain afterwards ;).

**Reply by steffann** [reply]
> adding workload display to netlab status show instance, potentially adding a special current instance to refer to the current directory

This is the option I have been looking for. So I guess this would be the most intuitive one for me. But I’m not normal, so listen to your own instinct! 🤣

**Reply by sdktr** [reply]
I don't grasp the different intended outputs as is implemented today. I'll have a look at what 'containerlab status' produces to get a feeling what you're proposing

**Reply by ipspace** [reply]
@sdktr Don't worry, it is confusing, I'll totally rebuild the command.

**Comment by ipspace** [comment]
With apologies to @sdktr, I will postpone this one until 1.8.1. It's time to push out everything accumulated so far so we can start collecting bugs ;)

**Comment by ipspace** [comment]
The new **netlab status** is out
